File: mycode/DSA-C/binary_search.c
#include<stdio.h>
#include<conio.h>
int binarysearch(int arr[],int n,int low,int high,int indx){
    int mid=(low+high)/2;
    if(low>high){
        return -1;
    }
if(arr[mid]==indx){
    return mid;
}
if(low<=high){
    if(arr[mid]<indx){
        binarysearch(arr,n,mid+1,high,indx);
    }else{
         binarysearch(arr,n,low,mid-1,indx);
    }
}
}
int main(){
    int arr[]={2,6,8,9};
    int n=sizeof(arr)/sizeof(int);
    printf("your arary is=%d",binarysearch(arr,n,0,n-1,9));
    return 0;
}

File: mycode/DSA-JAVA/binary_search.java
class biniarr
{
    public static int binarySearch(int arr[],int low,int high,int indx)
    {
        int mid=(low+high)/2;
        if(low>high)
        {
            return -1;
        }
    if(arr[mid]==indx)
    {
        return mid;
    }
    if(low<=high)
    {
        if(arr[mid]<indx)
        {
            binarySearch(arr,mid+1,high,indx);
        }else
        {
             binarySearch(arr,low,mid-1,indx);
        }
    }
    

}

public class binary_search
{
public static void main(String args[])
{
        int arr[]={2,6,8,9};
       // int n=sizeof(arr)/sizeof(int);
       int n=arr.length;
       int s=binarySearch(arr,0,n-1,9);
        System.out.println("your arary is="+s);

    }
   
}